### ISSUE TITLE
Allow strings containing no replacement strings

### ISSUE THREAD

**Issue by tversteeg** [museun/markings#4]
When I pass a string without any replacement strings it gives an error. Can this be removed or an option? I would like to use the replacement in a generic way over many strings, whether they contain the replacement string or not.

**Comment by museun** [comment]
Interesting. I have just implemented it, but I'd need some feedback on this usecase. https://github.com/museun/markings/commit/f53175622ec9ee95b5fc27381a2ad522aacd464f

This adds a method: `Template::is_empty()` to determine if `Template::apply()` will actually do any replacements.

```rust
let s = "some non-template string";
let template = Template::parse(s).unwrap();
assert!(template.is_empty());
let out = template.apply(&Args::new().build()).unwrap();
assert_eq!(s, out);
```

**Comment by tversteeg** [comment]
Thanks for the quick response!

This in fine for my use case. I'm only wondering if it wouldn't be more clean to just not throw any errors?

**Comment by museun** [comment]
A non-template input string wouldn't produce an error. The only errors returned in `Template::parse()` would be for mismatched braces, duplicate keys, etc.

Nor would `Template::apply()` return an error on an non-template string, it'll simply just move the input string out.

I will remove the `Error::EmptyTemplate` variant when I get ready to push a new version. (its unused now).

..or do you have a different concern?

**Comment by tversteeg** [comment]
No, that's perfect, I misunderstood your commit. Thanks again!

**Comment by museun** [comment]
I have published these changes as 0.2.0. Thanks for the suggestion!
